opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -334,11 +368,16 @@ impl<T: Number<T>> PrecomputedSum<T> {
             .has_no_value_attribute_value
             .swap(false, Ordering::AcqRel)
         {
+            let default = T::default();
+            let delta = self.value_map.no_attribute_value.get_value()
+                - *reported.get(&vec![]).unwrap_or(&default);
+            new_reported.insert(vec![], self.value_map.no_attribute_value.get_value());
```

Comment:
We should be storing the computed `delta` value here and not do another atomic read. Otherwise, we risk storing an incorrect value from another update thread.

Reply:
Good catch. The `self.value_map.no_attribute_value` fetched once in line 372  should be used as `new_reported` value instead of fetching again.